crates/chia-consensus/src/spendbundle_conditions.rs

```diff
@@ -94,6 +93,13 @@ pub fn run_spendbundle(
             clvm_cost,
             constants,
         )?;
+
+        let Some(ref mut spend) = ret.spends.last_mut() else {
+            panic!("process_single_spend() failed to add a spend");
+        };
```

Comment:
Hmm this looks weird, `process_single_spend` already returns a `Result` so by reaching this line it's reasonable to assume we added a spend. Isn't it better to make its success path return the spend (instead of unit) and use it here?

Reply:
this statement *does* assume we added the spend (and panics otherwise). I'm hesitant to always make a copy of the spend just to use it rarely (i.e. `ELIGIBLE_FOR_DEDUP` when `COMPUTE_FINGERPRINT` is specified).

Reply:
we can't return a mutable reference from `process_single_spend()`, since that would violate borrowing rules. The object is already referenced by `ret.spends`